src/ngraph/op/util/index_reduction.hpp

```diff
@@ -27,13 +32,21 @@ namespace ngraph
             class IndexReduction : public Op
             {
             public:
-                size_t get_reduction_axis() const { return m_axis; }
-                element::Type get_index_element_type() const { return m_index_element_type; }
+                IndexReduction();
```

Comment:
Probably want to change the ctors here to `protected`, like you did in `BinaryElementwiseArithmetic`.

Reply:
done